src/Translator/src/TranslationsDumper.php

```diff
@@ -189,13 +186,14 @@ private function generateConstantName(string $translationId): string
     {
         static $alreadyGenerated = [];
 
+        $translationId = s($translationId)->ascii()->snake()->upper()->replaceMatches('/^(\d)/', '_$1')->toString();
```

Comment:
If you have an ascii string here at one point, you'be probably quicker to compute the upper snake yourself, as the String implementation does some form of camel -> snake 

(cf https://github.com/twigphp/Twig/blob/3.x/src/Util/CallableArgumentsExtractor.php and the whole PR about it)

Reply:
I've just did a quick test with:
```php
$translationId = preg_replace(['/([A-Z]+)([A-Z][a-z])/', '/([a-z0-9])([A-Z])/'], '\1_\2', $translationId);
$translationId = str_replace(['.', '-'], '_', $translationId);
$translationId = strtoupper($translationId);

Reply:
Damn ... :( This string implementation has really too many impacts :( 

Ignore my comment then 😢

Reply:
Yeah that's not an easy topic 😅